applications/StructuralMechanicsApplication/custom_processes/set_automated_initial_variable_process.cpp

```diff
@@ -0,0 +1,132 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:		 BSD License
+//					 license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Luis Antonio Goncalves Junior
+//                   Alejandro Cornejo
+//
+
+#include "custom_processes/set_automated_initial_variable_process.h"
+#include "utilities/parallel_utilities.h"
+#include "utilities/math_utils.h"
+#include "custom_utilities/constitutive_law_utilities.h"
+#include "structural_mechanics_application_variables.h"
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/model_part.h"
+
+namespace Kratos
+{
+SetAutomatedInitialVariableProcess::SetAutomatedInitialVariableProcess(
+    ModelPart& rThisModelPart,
+    Parameters ThisParameters
+    ):mrThisModelPart(rThisModelPart),
+      mThisParameters(ThisParameters)
+{
+    mThisParameters.ValidateAndAssignDefaults(GetDefaultParameters());
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void SetAutomatedInitialVariableProcess::ExecuteInitialize()
+{
+    
+    KRATOS_TRY
+
+    const array_1d<double, 3> hole_generatrix_axis = mThisParameters["hole_generatrix_axis"].GetVector();
+    KRATOS_ERROR_IF(MathUtils<double>::Norm3(hole_generatrix_axis) < std::numeric_limits<double>::epsilon()) << "The hole generatrix axis has norm zero" << std::endl;
+    
+    const array_1d<double, 3> hole_generatrix_point = mThisParameters["hole_generatrix_point"].GetVector();  
+
+    array_1d<double, 3> normalized_generatrix_vector;   
+    normalized_generatrix_vector[0] = (hole_generatrix_axis[0] * hole_generatrix_axis[0]) / std::sqrt(hole_generatrix_axis[0] * hole_generatrix_axis[0] + hole_generatrix_axis[1] * hole_generatrix_axis[1] + hole_generatrix_axis[2] * hole_generatrix_axis[2]);
+    normalized_generatrix_vector[1] = (hole_generatrix_axis[1] * hole_generatrix_axis[1]) / std::sqrt(hole_generatrix_axis[0] * hole_generatrix_axis[0] + hole_generatrix_axis[1] * hole_generatrix_axis[1] + hole_generatrix_axis[2] * hole_generatrix_axis[2]);
+    normalized_generatrix_vector[2] = (hole_generatrix_axis[2] * hole_generatrix_axis[2]) / std::sqrt(hole_generatrix_axis[0] * hole_generatrix_axis[0] + hole_generatrix_axis[1] * hole_generatrix_axis[1] + hole_generatrix_axis[2] * hole_generatrix_axis[2]);
+
+    const double hole_radius_offset = mThisParameters["hole_radius_offset"].GetDouble();
+
+    block_for_each(mrThisModelPart.Elements(), [&](Element &rElement) {
+
+        const array_1d<double, 3> element_centroid = rElement.GetGeometry().Center();
```

Comment:
```suggestion
        const array_1d<double, 3>& r_element_centroid = rElement.GetGeometry().Center();
```

no need for the copy if it is anyway `const`